x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -0,0 +1,202 @@
+    private static final ParseField INDICES_FIELD = new ParseField("indices");
+
+    static {
+        PARSER.declareString(constructorArg(), NAME_FIELD);
+        PARSER.declareStringArray(constructorArg(), INDICES_FIELD);
+    }
+
+    /**
+     * Parses a {@link Engine} from its {@param xContentType} representation in bytes.
+     *
+     * @param resourceName The name of the resource (must match the {@link Engine} name).
+     * @param source The bytes that represents the {@link Engine}.
+     * @param xContentType The format of the representation.
+     *
+     * @return The parsed {@link Engine}.
+     */
+    public static Engine fromXContentBytes(String resourceName, BytesReference source, XContentType xContentType) {
+        try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+            return Engine.fromXContent(resourceName, parser);
+        } catch (IOException e) {
+            throw new ElasticsearchParseException("Failed to parse the get result", e);
+        }
+    }
+
+    /**
+     * Parses an engine through the provided {@param parser}.
+     *
+     * @param resourceName The name of the resource (must match the {@link Engine} name).
+     * @param parser The {@link XContentType} parser.
+     *
+     * @return The parsed {@link Engine}.
+     */
+    public static Engine fromXContent(String resourceName, XContentParser parser) throws IOException {
+        final Engine engine = PARSER.parse(parser, null);
+        if (engine.name().equals(resourceName) == false) {
+            throw new IllegalStateException(
+                "The name of the engine must match the resource name, got " + engine.name() + " and " + resourceName
+            );
+        }
+        return engine;
+    }
+
+    /**
+     * Converts the {@link Engine} to XContent.
+     *
+     * @return The {@link XContentBuilder} containing the serialized {@link Engine}.
+     */
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field("name", name);
+        builder.field("indices", indices);
+        builder.endObject();
+        return builder;
+    }
+
+    /**
+     * Returns the name of the {@link Engine}.
+     *
+     * @return The name of the {@link Engine}.
+     */
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Returns the list of indices targeted by the {@link Engine}.
+     *
+     * @return The list of indices targeted by the {@link Engine}.
+     */
+    public String[] indices() {
+        return indices;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Engine engine = (Engine) o;
+        return name.equals(engine.name) && Arrays.equals(indices, engine.indices);
```

Comment:
Note: This will require that the indices always have the same order. Otherwise, out-of-order indices and the same name will result in a mismatch. 

We could mitigate this by sorting the indices before comparing, or by using something like [Collections.containsAll()](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#containsAll-java.util.Collection-) if we are concerned with that level of equivalency. WDYT?

Reply:
Good catch, I pushed https://github.com/elastic/elasticsearch/pull/93792/commits/d633f8fb1e79ed2207c1597b2db38e6b384c7bb8 to sort the indices when the Engine is created.